### PR TITLE
Remove 'No Restart' label, instead show 'Requires Restart' where necessary

### LINKED ISSUE #3080: Hide restartless, focus on requiring restart
As e10s rolls out, users will forced to get a restart to move between the e10s cohort and not. Unfortunately we can't detect on AMO if they are in the cohort. The main suggestion has been to hide the restartless flag from AMO.

screenshot 2016-07-12 11 01 27

It appears on the add-ons details page and search results, maybe more.

There's a restartless tag: https://addons.mozilla.org/en-US/firefox/tag/restartless that gets added to add-ons, but we really don't advertise that too much. Perhaps we could add a message to that page to say "it may not be restartless".

### PR DESCRIPTION
**Pull request by diox**
Fix #3080.

The `no_restart` boolean field on File is kept as-is for now to avoid a migration and even more code changes.

### CODE REVIEW

File: src/olympia/tags/cron.py [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-from olympia.tags.models import AddonTag, Tag
-
-
-@cronjobs.register
-def tag_jetpacks():
-    # A temporary solution for singling out jetpacks on AMO.  See bug 580827
-    tags = (
-        ('jetpack', {
-            '_current_version__files__jetpack_version__isnull': False
-        }),
```
Comment (EnTeQuAk) [inline src/olympia/tags/cron.py]: Are you sure we don't want to tag `jetpack` addons anymore?

**Comment by EnTeQuAk** [comment]
r+wc

I'm not sure if we really want to remove tagging `jetpack` addons on AMO, I believe this is being used but I could very well confuse it with something else.

File: src/olympia/tags/cron.py (hunk 1 above)
Comment (diox) [inline src/olympia/tags/cron.py]: According to @andym on IRC, yes. For the loz: this cron was actually not used at all (though I also removed the tagging that was done at upload time).

Comment (EnTeQuAk) [inline src/olympia/tags/cron.py]: cc'ing @wagnerand regarding the tagging during upload time just to be extra sure. I remember hearing something that some tags are relevant for review purposes but maybe that's just relevant in validation messages but I'm not sure.

Comment (diox) [inline src/olympia/tags/cron.py]: Editors have various flags, including one for jetpack add-ons, but it's independent of the tag - I kept the jetpack flag for editors, and removed the restartless, replaced it with "requires restart" as per the frontend.

**Pull request merged**